### DISCUSSION TITLE
Anyway to speed up the Azure segment, takes about 800ms for me

### DISCUSSION THREAD

**Discussion by danijam** [JanDeDobbeleer/oh-my-posh#1217] [Troubleshoot] [answered]
Is there anything I can do to speed up the Azure segment?

I work daily with Azure from the CLI and having the context of the subscription I'm in is very valuable.

When I run Write-PoshDebug the Azure segment is always in the 700-800ms.

I'm running 6.2.5 of oh-my-posh

**Comment by JanDeDobbeleer** [comment, marked as answer]
@danijam yes, create a bug report for the AZ CLI 🐛, it's their tool that's slow, not oh-my-posh. What you could do is add that segment as a [tooltip](https://ohmyposh.dev/docs/beta#tooltips) rather than in your prompt at all times.

**Reply by danijam** [reply]
Thanks, on it!

I like the tooltip functionality, will have to try that out.

I'm right in saying that the az segment is using the az cli and not powershell? The command it is invoking is `az account show`? based on this? https://github.com/JanDeDobbeleer/oh-my-posh/blob/2733865edf050e4e47f899cdb0bec45916a242ff/src/segment_az.go#L89

with `measure-command { az account show }` reports a fairly consistent 500ms!!

**Reply by danijam** [reply]
Raised https://github.com/Azure/azure-cli/issues/20314

**Reply by JanDeDobbeleer** [reply]
What you could also try is enable fetching the information from the PowerShell module by setting `$env:AZ_ENABLED = $true` in your `$PROFILE` (doesn't matter where, we pick it up).

**Reply by danijam** [reply]
@JanDeDobbeleer Loving the update in https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v6.42.0 az segment now is basically ~1ms :)
